workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -252,8 +248,8 @@ internal class WorkspaceServiceImpl(
       overwrite: Boolean,
       destination: String?
   ): WorkspaceFile {
-    if (destination?.contains("..") == true) {
-      throw IllegalArgumentException("Invalid destination: '$destination'. '..' is not allowed")
+    require(destination?.contains("..") != true) {
```

Comment:
Is the inverted logic still correct when `destination` is `null`. It was not an issue before but now it will trigger the error, unless `?.` does not behave like I think.